### PR TITLE
PRO-1754 PRO-1857 PRO-1868 parked properties need to work in all locales, determine locale via prefix and domain, various fixes

### PR DESCRIPTION
**Pull request by boutell**
This PR should not be reviewed and approved until PRO-1754 is approved.

### CODE REVIEW

File: modules/@apostrophecms/i18n/index.js [hunk 1]
```diff
@@ -227,6 +324,15 @@ module.exports = {
           self.getComponentName('localizeModal', 'AposI18nLocalize'),
           { moduleName: self.__meta.name }
         );
+      },
+      setPrefixUrls(req) {
+        // For bc, req.baseUrl is always set, to a best guess if baseUrl is not configured.
+        // When falling back, req.hostname is used if trustProxy is active, otherwise the
+        // Host header to allow port numbers in dev
+        req.baseUrl = self.apos.page.getBaseUrl(req) || (req.protocol + '://' + (self.options.trustProxy && req.get('X-Forwarded-Host')) ? req.hostname : req.get('Host'));
```
Comment (abea) [inline modules/@apostrophecms/i18n/index.js:332]: I'd prefer you get the hostname as a step before to make this easier to parse. Then it could be cleanly written with a template literal.

File: modules/@apostrophecms/page-type/index.js [hunk 2]
```diff
@@ -323,7 +323,14 @@ module.exports = {
           return self.apos.page.insert(_req, doc.aposLastTargetId.replace(':draft', ':published'), doc.aposLastPosition, published, options);
         } else if (!doc.level) {
           // Insert the home page
-          return self.apos.doc.db.insertOne(_req, published, options);
+          Object.assign(published, {
+            path: doc.path,
```
Comment (abea) [inline modules/@apostrophecms/page-type/index.js:327]: Why this change?

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/express/index.js [hunk 3]
```diff
@@ -630,10 +630,6 @@ module.exports = {
       // set the `baseUrl` option for Apostrophe.
 
       addAbsoluteUrlsToReq(req) {
```
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/express/index.js]: This internal API has been made obsolete by the new `req.prefix` feature, to which its logic was moved. I've removed it properly.

File: modules/@apostrophecms/page-type/index.js (hunk 2 above)
Comment (boutell) [reply modules/@apostrophecms/page-type/index.js:327]: `published` at this point only has schema properties. If we don't bring over this information the home page will not replicate properly. It's different for non-home pages because we can use `page.insert` for them.

File: modules/@apostrophecms/i18n/index.js [hunk 4]
```diff
@@ -329,7 +329,8 @@ module.exports = {
         // For bc, req.baseUrl is always set, to a best guess if baseUrl is not configured.
         // When falling back, req.hostname is used if trustProxy is active, otherwise the
         // Host header to allow port numbers in dev
-        req.baseUrl = self.apos.page.getBaseUrl(req) || (req.protocol + '://' + (self.options.trustProxy && req.get('X-Forwarded-Host')) ? req.hostname : req.get('Host'));
+        const host = (self.options.trustProxy && req.get('X-Forwarded-Host')) ? req.hostname : req.get('Host');
```
Comment (abea) [inline modules/@apostrophecms/i18n/index.js:332]: Blessed.

**Review by abea — CHANGES_REQUESTED** [review]
